modules/src/mock_context.rs

```diff
@@ -74,6 +83,35 @@ impl MockContext {
         Self { clients, ..self }
     }
 
+    /// Similar to `with_client`, this function associates a client record to this context, but
```

Comment:
Shouldn't `with_client()` then be a wrapper call to this function with `None`s where needed?

Reply:
Good idea https://github.com/informalsystems/ibc-rs/pull/305/commits/a6d3d2e5011cf4bd8aab4d5e226107ce847bbd46 !